public/src/components/ToastManager.jsx

```diff
@@ -0,0 +1,198 @@
+import { useState, useRef } from 'react';
+import * as React from 'react';
+import { PropTypes } from 'prop-types';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faCheckCircle, faExclamationTriangle, faTimes,
+} from '@fortawesome/free-solid-svg-icons';
+
+const getCSSForFlavor = (flavor) => {
+  const flavors = {
+    success: `
+      background-color: #386C0B;
+    `,
+    error: `
+      background-color: #EF6F6C;
+    `,
+  };
+  return flavors[flavor];
+};
+
+export const ToastContext = React.createContext();
+const ToastRack = styled.div`
+  z-index: 1;
+  position: fixed;
+  right: 0;
+  width: 250px;
+  bottom: 0;
+  padding: 1em;
+`;
+
+const StyledToast = styled.div`
+  @keyframes styledtoast {
+    0% {
+      transform: translateY(20px)
+    }
+    100% {
+      transform: translateY(0px)
+    }
+  }
+  @keyframes cancelled {
+  0% {
+    transform: translateX(0px);
+  }
+  90% {
+    transform: translateX(0px);
+  }
+  100% {
+    transform: translateX(300px);
+  }
+}
+  @keyframes immediatecancel {
+  0% {
+    transform: translateX(0px);
+  }
+  100% {
+    transform: translateX(300px);
+  }
+}
+
+  display: flex;
+  border: 1px solid #54457F;
+  width: 250px;
+  cursor: pointer;
+  animation: styledtoast .3s ease-out, cancelled 4.5s ease-out .3s;
+  animation-fill-mode: forwards;  
+  border-radius: .25em;
+  background-color: white;
+  box-shadow: 0px 4px 3px 3px rgba(0,0,0, .1);
+  & + & {
+    margin-top: 1em;
+  }
+
+  & .toast__alert {
```

Comment:
You really shouldn't be using classnames like this in styled components. The whole point of styled components is to define units of experience and then use them as references to avoid having to rely on strings (classes) to tie things together

Reply:
Fixed.